docs/reference/pyboard.py.rst

```diff
@@ -0,0 +1,128 @@
+.. _pyboard_py:
+
+The pyboard.py tool
+===================
+
+This is a standalone Python tool that runs on your PC that provides a way to:
+
+* Quickly run a Python script or command on a MicroPython device. This is useful
+* while developing MicroPython programs to quickly test code without needing to
+* copy files to/from the device.
+
+* Access the filesystem on a device. This allows you to deploy your code to the
+* device (even if the board doesn't support USB MSC).
+
+Despite the name, ``pyboard.py`` works on all MicroPython ports that support the
+raw REPL (including STM32, ESP32, ESP8266, NRF).
+
+You can download the latest version from `GitHub
+<https://github.com/micropython/micropython/blob/master/tools/pyboard.py>`_. The
+only dependency is the ``pyserial`` library which can be installed from PiPy or
+your system package manager.
+
+Running ``pyboard.py --help`` gives the following output:
+
+.. code-block:: text
+
+    usage: pyboard [-h] [--device DEVICE] [-b BAUDRATE] [-u USER]
+                   [-p PASSWORD] [-c COMMAND] [-w WAIT] [--follow] [-f]
+                   [files [files ...]]
+
+    Run scripts on the pyboard.
+
+    positional arguments:
+      files                 input files
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      --device DEVICE       the serial device or the IP address of the
+                            pyboard
+      -b BAUDRATE, --baudrate BAUDRATE
+                            the baud rate of the serial device
+      -u USER, --user USER  the telnet login username
+      -p PASSWORD, --password PASSWORD
+                            the telnet login password
+      -c COMMAND, --command COMMAND
+                            program passed in as string
+      -w WAIT, --wait WAIT  seconds to wait for USB connected board to become
+                            available
+      --follow              follow the output after running the scripts
+                            [default if no scripts given]
+      -f, --filesystem      perform a filesystem action
+
+Running a command on the device
+-------------------------------
+
+This is useful for testing short snippets of code, or to script an interaction
+with the device.::
+
+    $ pyboard.py --device /dev/ttyACM0 -c 'print(1+1)'
+    2
+
+Running a script on the device
+------------------------------
+
+If you have a script, ``app.py`` that you want to run on a device, then use::
+
+    $ pyboard.py --device /dev/ttyACM0 app.py
+
+Note that this doesn't actually copy app.py to the device's filesystem, it just
+loads the code into RAM and executes it. Any output generated by the program
+will be displayed.
```

Comment:
"If the program `app.py` does not finish then you'll need to stop `pyboard.py`, eg with ctrl-C.  The program `app.py` will still continue to run on the MicroPython device.